include/wx/localedefs.h

```diff
@@ -108,6 +108,83 @@ struct WXDLLIMPEXP_BASE wxLanguageInfo
     const char* TrySetLocale() const;
 };
 
+// ----------------------------------------------------------------------------
+// wxLocaleIdent: allows to fully identify a locale under all platforms
+// ----------------------------------------------------------------------------
+
+class WXDLLIMPEXP_BASE wxLocaleIdent
+{
+public:
+    // Leave language empty
+    wxLocaleIdent() { }
```

Comment:
What does "empty language" mean? Some of the code here suggests it means the default, but the docs don't spell it out.

Reply:
Yes, it is "the default", I'll have to explain it better.

This is another place where I hesitated for a long time: on one hand, a "default" identifier is not really useful, especially because it's not clear whether it's a system default, user default or something else. But OTOH this is a value-like class, so it would be strange to not have a default ctor in it...

Reply:
> This is another place where I hesitated for a long time: on one hand, a "default" identifier is not really useful, especially because it's not clear whether it's a system default, user default or something else.

With (hopefully) better understanding now of what you're doing with this class, I'd say the default value semantics should definitely be "user default" and that a distinct default value is *necessary*. If a) wxLocaleIdent is used internally and b) it cannot, by design, represent all languages/locales, then a distinct way to represent user default is necessary.

Reply:
OTOH we already have `wxUILocale::CreateUserDefault()` and having `wxUILocale(wxLocaleIdent())` do the same thing doesn't look like such a great idea.

I think I'd rather make default-constructed `wxLocaleIdent` invalid, i.e. assert that it's not empty in `wxUILocale` ctor. What do you think?